monkey/common/utils/secret_variable.py

```diff
@@ -0,0 +1,16 @@
+from typing import Any
+
+from pydantic.types import SecretStr
+
+from common.utils.i_secret_variable import ISecretVariable
+
+
+class SecretVariable(ISecretVariable):
+    def __init__(self, secret_value: Any):
+        if isinstance(secret_value, str):
+            self._secret_value = SecretStr(secret_value)
+        else:
+            raise NotImplementedError("SecretVariable only supports string values.")
+
+    def get_secret_value(self) -> Any:
+        return self._secret_value.get_secret_value()
```

Comment:
Why aren't we just doing this?

```python
>>> from pydantic import SecretStr
>>>
>>> class OTP(SecretStr):
...     pass
...
>>>
>>> otp = OTP("1q2w3e")
>>> otp
SecretStr('**********')
>>> otp.get_secret_value()
'1q2w3e'
```

Reply:
This couples the OTP to pydantic, because in this case OTP will inherit all of the interface of the SecretStr (we only need `get_secret_value()`). What we want is to hide the value of a variable from logs. When another secret type of bytes or dict comes up, how are we going to hide it?